### ISSUE TITLE
Support bundling DLLs from $OUT_DIR

### ISSUE THREAD

**Issue by lukexor** [axodotdev/cargo-dist#542]
I'm building a binary distributable that relies on a library crate, which itself dynamically links to SDL2. For windows builds, the build.rs of that library sticks the correct x86_64 or i686 DLLs in the $OUT_DIR location for cargo so they will be present alongside the binary.

Currently the `include` feature would work for this, but I have no way of naming $OUT_DIR in Cargo.toml since it's not set until build time and globs are not supported.

I did see #194 but this request is a bit more limited in scope as these are build artifacts, not a generic random path outside $CARGO_MANIFEST_DIR

**Comment by Gankra** [comment]
makes perfect sense! I wonder if the linkage checker can detect this so we know to automatically include the dll 🤔

**Comment by lukexor** [comment]
That would be great! Currently, I'm not sure the best work-around.

I'm trying to learn the WIX toolset and I could modify the .wix file to include each file (or copy the dlls into my project instead of relying on $OUT_DIR and using Heat to collect the whole directory) - but then I lose out on the updating features of cargo dist init.

**Comment by Gankra** [comment]
Oh! Thanks for the additional context that you specifically interested in MSIs here -- some additional work may also be needed to teach MSIs to do additional includes: https://github.com/axodotdev/cargo-dist/issues/543

**Comment by lukexor** [comment]
I was able to get a decently working solution by just setting `allow-dirty = ["msi"]` and then modifying the `main.wxs` file to add the necessary DLL dependencies. This also allowed me to customize the installation images and have it launch the application after install.

The only downsides are having to manually manage any new wix features cargo dist releases, or migrating to wix 4 whenever that happens.

**Comment by mistydemeo** [comment]
> there seems to be no way to customize the dependencies for windows like there is for homebrew and apt.

It's possible to customize the dependencies that are installed during the build, but it's not currently possible to get the install script to install external packages alongside your software.

**Comment by lukexor** [comment]
> > there seems to be no way to customize the dependencies for windows like there is for homebrew and apt.
> 
> It's possible to customize the dependencies that are installed during the build, but it's not currently possible to get the install script to install external packages alongside your software.

Like being able to add dlls and customize the install dir? Currently powershell sticks the binary in .cargo/bin but stuffing dlls in there isn't ideal. The app I'm distributing requires SDL2 libraries.

**Comment by Gankra** [comment]
Yeah including more than just the one file should require you to [set a custom install-path](https://opensource.axo.dev/cargo-dist/book/reference/config.html#install-path), but we need a clear story for how to specify whether your custom install path is ok to dump More Files into.

(A new config for saying "yep it's fine" and changes to the ps1 and sh installers to detect that config and unpack more from the archive at install-time.)
